### ISSUE TITLE
Calling a function named `for` in `case` or `if` messes up indentation of body

### ISSUE THREAD

**Issue by utkarshkukreti** [elixir-lang/emacs-elixir#323]
I have a function named `for` in a module and calling it in `case` or `if` from another module is messing up the indentation of the whole body.

Input / Expected indentation:

```elixir
defmodule Whois do
  def lookup1(domain) do
    case Server.for(domain) do
      {:ok, server} -> server
      :error -> {:error, :unsupported}
    end
  end

  def lookup2(domain) do
    if Server.for(domain) do
      :ok
    else
      :error
    end
  end
end
```

Actual, after selecting all text and calling `indent-region`:

```elixir
defmodule Whois do
  def lookup1(domain) do
    case Server.for(domain) do
          {:ok, server} -> server
          :error -> {:error, :unsupported}
        end
  end

  def lookup2(domain) do
    if Server.for(domain) do
          :ok
        else
          :error
        end
  end
end
```

```
GNU Emacs 24.5.7 (x86_64-apple-darwin13.4.0, Carbon Version 157 AppKit 1265.21) of 2015-09-28 on Atago.local
Elixir-Mode version: 2.2.9 (package: 20160406.723)
```

**Comment by gausby** [comment]
I've added some failing tests. I've split them up in a if-test and a case-test.

**Comment by tonini** [comment]
@utkarshkukreti Thanks for the report. The issue is fixed.

You need to wait until melpa.org build the new `elixir-mode` package from master before you install it.

**Comment by utkarshkukreti** [comment]
@tonini Thank you! I can confirm this is fixed in `Elixir-Mode version: 2.2.9 (package: 20160412.1223)`.
